akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -219,6 +237,41 @@ private[http2] class Http2ServerDemux(http2Settings: Http2CommonSettings, initia
         }
       })
 
+      /**
+       * Tune this peer to enforce the settings configured from this peer.
+       * @return TODO: it's a Boolean but I don't think it has to be.
+       */
+      private def enforceSettings(settings: immutable.Seq[Setting]): Boolean = {
+        var settingsAppliedOk = true
+
+        settings.foreach {
+          case Setting(Http2Protocol.SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, value) =>
+          // TODO: Http2Compliance.compliesWithMaxConcurrentStreams should only be enabled once
+          //  we get this SETTINGS_ACK.
+        }
+
+        settingsAppliedOk
+      }
+
+      protected override def checkMaxConcurrentStreamsCompliance(currentConcurrentStreams: => Int): Unit = {
+        if (currentConcurrentStreams >= maxConcurrentStreams) {
+          // 5.1.2 An endpoint that receives a HEADERS frame that causes its advertised
+          // concurrent stream limit to be exceeded MUST treat this as a stream error
+          // (Section 5.4.2) of type PROTOCOL_ERROR or REFUSED_STREAM.
+          pushGOAWAY(
```

Comment:
Didn't we want to use `RST_STREAM` with `REFUSED_STREAM` instead? Or is that for a follow up PR?

Reply:
My bad. I overlooked the change.